crypto/buffer/buffer.c

```diff
@@ -141,7 +145,6 @@ int BUF_MEM_grow_clean(BUF_MEM *str, size_t len)
         return (len);
     }
     if (str->max >= len) {
-        memset(&str->data[str->length], 0, len - str->length);
```

Comment:
Hi kroeckx, Thanks for your comment. 

As per my knowledge, the different between `BUF_MEM_grow` and `BUF_MEM_grow_clean` is not providing memsetted buffer. Rather `BUF_MEM_grow_clean' should do `CRYPTO_realloc_clean`, that means during expansion it should do `OPENSSL_cleanse` before releasing old buffer.

So as per my knowledge both the function should return memsetted buffers.